scripts/sniffer-testsuite.test

```diff
@@ -77,16 +83,42 @@ fi
 if test $RESULT -eq 0 && test $has_rsa == yes && test $has_tlsv12 == yes && test $has_static_rsa == yes
 then
     echo -e "\nStaring snifftest on sniffer-ipv6.pcap...\n"
-    ./sslSniffer/sslSnifferTest/snifftest ./scripts/sniffer-ipv6.pcap ./certs/server-key.pem ::1 11111
+    ./sslSniffer/sslSnifferTest/snifftest -pcap ./scripts/sniffer-ipv6.pcap -key ./certs/server-key.pem -server ::1 -port 11111
 
     RESULT=$?
     [ $RESULT -ne 0 ] && echo -e "\nsnifftest (ipv6) failed\n" && exit 1
 fi
 
+#  TLS v1.2 sniffer keylog file test: runs sniffer on pcap and associated keylog file and compares decrypted traffic with known good output.
+#  To regenerate the known good output, run `scripts/sniffer-gen.sh` to regenerate the pcap and keylog file, then run the sniffer on it
+#  with the same arguments as in the test belowl, but redirect output to `./scripts/sniffer-tls12-keylog.out`.
+if test $RESULT -eq 0 && test $has_tlsv13 == yes && test $has_keylog == yes
+then
+    TMPFILE=$(mktemp)
+    RESULT=$?
+    [ $RESULT -ne 0 ] && echo -e "\nsnifftest keylog test failed: unable to create tmpfile\n" && rm $TMPFILE && exit 1
+
+    ./sslSniffer/sslSnifferTest/snifftest \
```

Comment:
Failure on the sniffer-testsuite.test with 

```
./configure --enable-sniffer CFLAGS="-DWOLFSSL_SNIFFER_KEYLOGFILE -DTHREADED_SNIFFTEST" && make
...
./scripts/sniffer-testsuite.test
...
Found duplicate client random value in keylog file. Rejecting.
Found duplicate client random value in keylog file. Rejecting.
Found duplicate client random value in keylog file. Rejecting.
Found duplicate client random value in keylog file. Rejecting.
Found duplicate client random value in keylog file. Rejecting.
Found duplicate client random value in keylog file. Rejecting.

snifftest keylog test failed: snifftest returned 1
```

Reply:
when I run it I see that error occasionally, but also see a bunch of the other non-keylog snifftests failing and segfaulting. This behavior also exists on `master`, so I think something is just wrong with the multi-threaded snifftest in general. I'll investigate.

Reply:
Please don't spend time on the threaded issue for now. The feature was added not long ago by @lealem47. He should be the one to followup with it when he has time. Please focus on the TLS v1.3 key logging for now. Thanks